### PR TITLE
Refactoring getClosestPlotData()'s API to the new, simpler version.

### PR DESCRIPTION
**Pull request by istarnx**

### CODE REVIEW

**Comment by istarnx** [comment]
Oh, by the way, I added getClosestPlotData(), a public method, to the bottom of linePlot.ts, before the public Hover api stuff. This seemed weird. We're usually placing public stuff at the top if files, right?

File: src/components/plots/abstractPlot.ts [hunk 1]
```diff
@@ -516,30 +516,17 @@ export module Plot {
...
+     * Retrieves the closest PlotData across all datasets, where distance is defined to be
+     * the Euclidiean norm.
      *
-     * @param {Point} queryPoint The point to query from
-     * @param {number} withinValue Will only return plot data that is of a distance below withinValue
-     *                             (default = Infinity)
-     * @param {string | string[]} datasetKeys The dataset(s) to retrieve the plot data from.
-     *                                        (default = this.datasetOrder())
-     * @returns {PlotData} The retrieved PlotData.
+     * @param {Point} queryPoint The point to which dataset points should be compared
```
Comment (bluong) [inline src/components/plots/abstractPlot.ts]: Let's change the doc possibly to 

Retrieves PlotData with the lowest distance, where distance is defined to be the Euclidiean norm.

File: src/components/plots/linePlot.ts [hunk 2]
```diff
@@ -182,6 +154,37 @@ export module Plot {
...
     }
 
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var closestDistanceSquared = Infinity;
+      var closestDatum: any;
+      var closestSelection: D3.Selection;
+      var closestPoint: Point;
+      var datasetKeys = this.datasetOrder();
+
+      datasetKeys.forEach((datasetKey: string) => {
```
Comment (bluong) [inline src/components/plots/linePlot.ts]: Minor, datasetKeys doesn't seem to be used anywhere.  Inline this.datasetOrder with the forEach?

**Comment by bluong** [comment]
When this is finished, we are going to want to document this behavior

**Comment by bluong** [comment]
This seems to be a simple removal of behavior.  Just fix up the docs

File: src/components/plots/linePlot.ts (hunk 2 above)
Comment (istarnx) [inline src/components/plots/linePlot.ts]: Good catch... this function is going to be rewritten in the next checkin anyway, so let's not worry about it. :)

File: src/components/plots/abstractPlot.ts (hunk 1 above)
Comment (istarnx) [inline src/components/plots/abstractPlot.ts]: Both of these make sense to me. I'll do 'em.

**Pull request merged**